### PR TITLE
Add test for ending up in wrong skip_next state on empty db

### PR DESCRIPTION
**Pull request by arj03**
First test should reproduce the bug seen in [replication scheduler](https://github.com/ssb-ngi-pointer/ssb-replication-scheduler/pull/5#issuecomment-1043197983)

The problem was that it was possible to set skip_next on an empty database so that the first value is actually skipped and never returned to the indexes.

### CODE REVIEW

File: stream.js [hunk 1]
```diff
@@ -139,14 +139,14 @@ Stream.prototype._resume = function () {
...
+  if (this.state === STREAM_STATE.INITIALIZING)
+    return // not ready yet
+
   if (!this.sink || this.sink.paused) {
     this.state = STREAM_STATE.PAUSED
     return
   }
 
-  if (this.state === STREAM_STATE.INITIALIZING)
-    return // not ready yet
```
Comment (arj03) [inline stream.js:148]: before what could happen is that you would trigger a resume and the sink was either no available or pasued. In this case we end up in state PAUSED. If you then resume again and sink is not paused, you will be allowed to continue and because the db is empty, you'll end up in the BLOCK_STATE.END_OF_STREAM where for a live stream we set skip_next true because we assume that you already got whatever the latest data was.

File: test/stream.js [hunk 2]
```diff
@@ -220,6 +220,40 @@ tape('live gt', function (t) {
...
+  var newLog = Log(filename1, {blockSize: 64*1024})
+
+  var sink = collect(function (err) {
+    if (err === 'tape-ended') return
+    else throw new Error('live stream should not end')
+  })
+  let ls = newLog.stream({ live: true, offsets: false, gt: -1 })
+  ls.pipe(sink)
+
+  setTimeout(() => {
```
Comment (arj03) [inline test/stream.js:238]: timeout to trigger ready

File: test/stream.js [hunk 3]
```diff
@@ -220,6 +220,40 @@ tape('live gt', function (t) {
...
+    sink.paused = true
+    ls.resume()
+    sink.paused = false
+    ls.resume()
```
Comment (arj03) [inline test/stream.js:242]: the resume dance that triggers the bug

**Comment by arj03** [comment]
While it always feels bad to have introduced a bug on the positive side:

1) There was no test of this before
2) The code is a lot easier to reason about, so even a tricky bug like that took a little more than an hour to fix

**Review by staltz — APPROVED** [review]
Thanks for finding this quickly!

I'll send a PR soon after this, to refactor tests (just for readability, avoiding 1 or 2 letter-long variables, etc)

**Comment by arj03** [comment]
Great, I'll hold up on making a release then :)

**Comment by staltz** [comment]
Actually, I think we can just release a patch, because a refactor in the tests shouldn't affect the release.

**Comment by arj03** [comment]
Sure, I'll release 3.1.4
